src/test/java/com/productdock/library/rental/service/RentalRecordApiTest.java

```diff
@@ -158,6 +167,35 @@ void shouldReturnBadRequest_whenReturningABookNotRentedByUser() throws Exception
                 .andExpect(status().isBadRequest());
     }
 
+    @Test
+    void shouldGetBookRecords() throws Exception {
+        givenAnyRentalRecord();
+
+        mockMvc.perform(get("/api/rental/record/" + FIRST_BOOK)
+                        .with(jwt().jwt(jwt -> jwt.claim("email", PATRON_1))))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.*").value(hasSize(2)))
+                .andExpect(jsonPath("$.[*].status",
+                        containsInAnyOrder(RentalStatus.RENTED.toString(), RentalStatus.RETURNED.toString())))
+                .andExpect(jsonPath("$.[*].email",
+                        containsInAnyOrder("default@gmail.com", "::email::")));
+
+    }
+
+    private void givenAnyRentalRecord() {
+        var returnedInteraction = defaultBookInteractionBuilder().userEmail("::email::").status(RentalStatus.RETURNED).build();
+        var rentedInteraction = defaultBookInteraction();
+        var bookInteractions = of(
+                rentedInteraction,
+                returnedInteraction
+        ).collect(toList());
+
+        var rentalRecord = defaultRentalRecordEntityBuilder().interactions(bookInteractions).build();
```

Comment:
This can be used like this, if @Singular is added to the RentalRecordEntityBuilder's collection
```
        var rentalRecord = defaultRentalRecordEntityBuilder()
                .interaction(rentedInteraction)
                .interaction(returnedInteraction)
                .build();
```